aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,767 @@
+
+        Parameters
+        ----------
+        lif: LifFile
+            The LifFile to get the ranges from
+        read_dims: Dict[str: int]
+            The list of locked dimensions
+
+        Returns
+        -------
+        Dict[Dimension: range]
+            These ranges can then be used to iterate through the specified YX images
+
+        """
+        if read_dims is None:
+            read_dims = {}
+
+        data_shape = LifReader._dims_shape(lif=lif)
+
+        # if S is in read_dims then use the specified value and the specified dims for that scene
+        if Dimensions.Scene in read_dims:
+            s_range = range(read_dims[Dimensions.Scene], read_dims[Dimensions.Scene] + 1)
+            s_dict = data_shape[s_range[0]]
+        else:
+            s_range = range(*data_shape[0][Dimensions.Scene])
+            s_dict = data_shape[0]
+
+        # map the dims over to ranges and if the dim is in read_dims make the range over the single dim
+        integrated_dims = {Dimensions.Scene: s_range}
+        for dim in [Dimensions.Time, Dimensions.Channel, Dimensions.SpatialZ]:
+            if dim in read_dims:
+                integrated_dims[dim] = range(read_dims[dim], read_dims[dim]+1)
+            else:
+                integrated_dims[dim] = range(*s_dict[dim])
+
+        return integrated_dims
+
+    @staticmethod
+    def get_pixel_type(meta: Element, scene: int = 0) -> np.dtype:
+        """
+        This function parses the metadata to assign the appropriate numpy.dtype
+
+        Parameters
+        ----------
+        meta: lxml.etree.Element
+            The root Element of the metadata etree
+        scene: int
+            The index of the scene, scenes could have different storage data types.
+
+        Returns
+        -------
+        numpy.dtype
+            The appropriate data type to construct the matrix with.
+
+        """
+        ############################
+        #
+        #  Due to the 12 bit values being stored in a uint16 the raw data is a little fussy to get the
+        #  contrast correct.
+        #
+        ############################
+        p_types = {8: np.uint8,
+                   12: np.dtype('<u2'),  # little endian uint16
+                   16: np.dtype('<u2'),  # little endian uint16
+                   32: np.dtype('<u4'),  # little endian uint32 ** untested
+                   64: np.dtype('<u8')   # little endian uint64 ** untested
+                   }
+        img_sets = meta.findall(".//Image")
+
+        img = img_sets[scene]
+        chs = img.findall(".//ChannelDescription")
+        resolution = set([ch.attrib['Resolution'] for ch in chs])
+        if len(resolution) != 1:
+            raise exceptions.InconsistentPixelType(f"Metadata contains two conflicting "
+                                                   f"Resolution attributes: {resolution}")
+
+        return p_types[int(resolution.pop())]
+
+    @staticmethod
+    def _get_item_as_bitmap(im_path: Path,
```

Comment:
Shouldn't this function be called `_get_array_from_offset`? Maybe that's nitpicky but seems odd to say `_get_item_as_bitmap` when you are returning an array.